### PR TITLE
Remove compose types.Dict alias

### PR DESCRIPTION
**Pull request by vdemeester**
It is just an alias type and make the code a little bit more complex and hard to use from outside `compose` package. It makes the `compose/interpolation` package indenpendent of the `compose/types` package (which is a good thing 👼)

/cc @dnephin @AkihiroSuda @thaJeztah @cpuguy83 

🦁

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Comment by thaJeztah** [comment]
SGTM. I do like `types.Dict` as it's descriptive, but I see where it complicates things 😅

File: cli/compose/loader/loader.go [hunk 1]
```diff
@@ -265,10 +265,10 @@ func transformHook(
 }
 
 // keys needs to be converted to strings for jsonschema
-// TODO: don't use types.Dict
+// TODO: don't use map[string]interface{}
```
Comment (dnephin) [inline cli/compose/loader/loader.go]: This TODO can be removed now

**Review by dnephin — APPROVED** [review]
LGTM

File: cli/compose/loader/loader.go (hunk 1 above)
Comment (vdemeester) [inline cli/compose/loader/loader.go]: ah lol yeah 👼 *auto-rename* 😛

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by icecrime — APPROVED** [review]
LGTM

**Review by chawlanikhil24 — APPROVED** [review]

**Pull request merged**